### ISSUE TITLE
Two factor authentication

### ISSUE THREAD

**Issue by danielrosehill** [mathesar-foundation/mathesar#3573]
Feel like this is really a must for any web-accessible solution that's providing a backend to potentially lots of sensitive data. 

I have mine behind Cloudflare Access, but I think if 2FA were implementable I'd feel comfortable having it a bit more exposed - which would in turn make it a lot easier to collaborate on management.

**Comment by seancolsen** [comment]
Thanks for this suggestion, @danielrosehill. I agree that 2FA would add significant value to the product.

Out of curiosity, what is your personal preference around the _type_ of 2FA. Most apps use SMS-based 2FA, which I personally don't like. I prefer authenticator-app-based 2FA, but I know it can sometimes be a barrier for less technically-inclined users who don't want to install an additional app. Then there are other things like FIDO and WebAuthn/Passkeys. If you have any opinions about these approaches, we'd love to hear them!

**Comment by dselfie** [comment]
Authenticator and/or biometric a la 1PW, stripe, etc.

**Comment by marksk1** [comment]
I agree that adding 2FA is kind of a must to use this in production - personally I prefer authenticator app
